### ISSUE TITLE
Disable Warning for complex numbers

### ISSUE THREAD

**Issue by NEGU93** [tensorflow/tensorflow#41557]
When running the following code:
```
import tensorflow as tf

x = tf.Variable(tf.complex([2., 2.], [2., 2.]))
with tf.GradientTape() as tape:
    y = tf.abs(tf.reduce_sum(x))**2
dy_dx = tape.gradient(y, x)
```
I get the warning message:
`WARNING:tensorflow:The dtype of the source tensor must be floating (e.g. tf.float32) when calling GradientTape.gradient, got tf.complex64`

According to [Introduction to Gradients and Automatic Differentiation](https://www.tensorflow.org/guide/autodiff#3_took_gradients_through_an_integer_or_string). This message is intended when the variable is not derivable. However, it does make sense to use the autodiff for complex numbers and this Warning message impacts my code negatively. I don't want to disable ALL warnings just for this.

This issue is related to [#32774](https://github.com/tensorflow/tensorflow/issues/32774) and [#30107 PR](Make tape only watch the tensor with the floating type).

**System information**
- TensorFlow version (you are using): 2.00
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
There is a warning in my opinion should not exist.

**Will this change the current api? How?**
Complex gradient will no longer display a Warning

**Who will benefit with this feature?**
Whoever is working with complex numbers.

**Comment by amahendrakar** [comment]
@NEGU93,
On running the code with [TF v2.0](https://colab.research.google.com/gist/amahendrakar/5ce6af79ad9ac754efeda05a6f7a44f5/41557-2-0.ipynb), I got a similar warning. However, looks like the issue is fixed in TF v2.2.

Running the code with [TF v2.2](https://colab.research.google.com/gist/amahendrakar/62c14e9b0f2852d42930bf95466947c7/41557.ipynb) I did not get any warning. Please find the attached gist. Thanks!

**Comment by NEGU93** [comment]
Ok, noted. I am using v2.0 because is the latest one added on Anaconda. I will see to create a new env and work with v2.2 then. Thanks!

**Comment by NEGU93** [comment]
True, I just updated to v2.2 and this problem does not exist anymore. Sorry!
